jujugui/static/gui/src/app/components/inspector/unit-list/unit-list.js

```diff
@@ -125,14 +125,21 @@ YUI.add('unit-list', function() {
       @param {String} action The action to apply to the units.
     */
     _handleUpdateUnits: function(action) {
-      var units = [];
-      var refs = this.refs;
-      var envResolved = this.props.envResolved;
+      let unitNames = [];
+      const units = this.props.units;
+      const refs = this.refs;
+      const envResolved = this.props.envResolved;
       Object.keys(refs).forEach(function (ref) {
-        var isInstance = ref.split('-')[0] === 'CheckListItem';
+        let isInstance = ref.split('-')[0] === 'CheckListItem';
         if (isInstance && refs[ref].state.checked) {
-          var unitName = ref.slice(ref.indexOf('-') + 1);
-          units.push(unitName);
+          let unitName = ref.slice(ref.indexOf('-') + 1);
+          unitNames.push(unitName);
+          const unit = units.filter((u) => {
```

Comment:
So you're looping through all the units here and then checking the `agent_status` of one of them with the same id? This will loop through ever unit so this should be more efficient:

```JavaScript
const unit = units.find(u => u.id === unitName);
```